src/main/webapp/mentors.js

```diff
@@ -0,0 +1,47 @@
+/**
+ * Get mentors from the mentor servlet.
+ */
+function getMentors() {
+  console.log("entering get mentors function/n");
+  let url = new URL('/mentors', location.protocol + '//' + location.hostname);
+
+  fetch(url).then(response => response.json()).then(mentors => {
```

Comment:
Look to add error handling to this fetch request in the future. If you add a `.catch((err) => [...])` after the fetch it should cover when there is an error making the request (like a network error). Take a look at [promise catches](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/catch) for more context. And then from within the response (before converting to json), you can add a check to determine if the request was successful (as determined by the response's [HTTP code](https://www.restapitutorial.com/httpstatuscodes.html)). This is a nice shortcut:
```
.then((response) => {
  if (response.ok) {
  [...]
  }
}
```
Where the response is okay if it is a 2xx code. The one thing that is a bit messy is what to do when the response is not okay. There are a bunch of ways you could handle this. If you handle the error from within that if-statement and don't return anything, it is the same as returning a resolved promise with a value of undefined, so it will get to the next then block with the json, but in this case the json will be undefined so you could then check if there was an error with something like:
```
if (json) {
[...]
}
```

Thanks for listening to my TED talk!

Reply:
Thanks for the advice and concrete examples!